### ISSUE TITLE
How to get extends "links" on doc pages

### ISSUE THREAD

**Issue by mathomp4** [Fortran-FOSS-Programmers/ford#422]
With the help of @ZedThree and a bit of excluding I was able to generate some documentation. Huzzah!

However, I find myself wanting *fancier* documentation. 😄 

Namely, if I look at the docs for GS2 here:

https://gyrokinetics.gitlab.io/gs2/lists/types.html

I see that, say, `ballstab_config_type` extends `abstract_config_type` and both of those have links. But in my docs, I only get the link on the first column, but not on the second column. Here's a screenshot (docs only exist on my laptop):

<img width="1175" alt="Screen Shot 2022-06-14 at 3 20 24 PM" src="https://user-images.githubusercontent.com/2982494/173671289-9da6c228-6662-45b3-83eb-287ea5469815.png">

You can see that `AbstractCollectiveDataMessage` extends `AbstractDataMessage` which itself extends `AbstractMessage`, but I don't have those snazzy links in the second column or in the page for the type itself.

Is there perhaps an option I need to pass to FORD? I tried looking at the [project file options](https://github.com/Fortran-FOSS-Programmers/ford/wiki/Project-File-Options), but nothing seems to stand out to my reading. I even found what I *think* is GS2's [configuration file](https://bitbucket.org/gyrokinetics/gs2/raw/5856fed635bb3d858f34e5a44fe467948e2974c4/docs/docs.config.md) and while I don't understand every option, nothing seems obvious.

Any ideas?

**Comment by haraldkl** [comment]
It should actually work. See for example https://geb.inf.tu-dresden.de/doxy/musubi/lists/types.html, though that is a little special, as it points to an external project, and links there. However, I believe for the sample in [test_data/external_project/external_project](https://github.com/Fortran-FOSS-Programmers/ford/tree/master/test_data/external_project/external_project) it also works and produces a link. Which version are on?

**Comment by mathomp4** [comment]
@haraldkl Yeah. I'm a bit baffled. But @ZedThree in https://github.com/Fortran-FOSS-Programmers/ford/issues/421#issuecomment-1156217398 did find that we might be having side-effects due to (at least) preprocessing issues. It's entirely possible the missing links are due to that? Not sure, but once I do more clean up, maybe it'll magically fix? We shall see...

**Comment by ZedThree** [comment]
I think #425 fixes this. Could you try the latest master and see if it works for you?

**Comment by mathomp4** [comment]
> I think #425 fixes this. Could you try the latest master and see if it works for you?

Woo hoo! It does! Thanks!

I guess the next thing for me (other than actually documenting the code 😄 ) is to figure out how to tune the graph generation. I tried these settings (based on [stdlib](https://github.com/fortran-lang/stdlib/blob/master/API-doc-FORD-file.md)):
```
graph: true
coloured_edges: true
graph_maxdepth: 5
graph_maxnodes: 250
```
and, well, it took over 3 hours to generate the documentation compared to ~10 minutes without the graphs. Maybe doc generation will be done at night... 😆

**Comment by haraldkl** [comment]
We use

```
graph_maxdepth: 4
graph_maxnodes: 32
```

In the project I linked earlier. That works quite nicely. Graphs with many more nodes anyway are hard to understand.
